### PR TITLE
[plugin-mobile-app] Add shooting strategy to take full screen screenshot

### PR DESCRIPTION
**Pull request by IBrun**

### CODE REVIEW

File: vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java [hunk 1]
```diff
@@ -47,24 +54,31 @@ public MobileAppAshotFactory(ScreenshotCropper screenshotCropper, MobileAppWebDr
...
-
         ShootingStrategy strategy;
-        if (appendBottomNavigationBarOnAndroid && mobileAppWebDriverManager.isAndroid())
-        {
-            strategy = getStrategyBy(strategyName).getDecoratedShootingStrategy(ShootingStrategies.simple());
-            int statusBarSize = mobileAppWebDriverManager.getStatusBarSize();
-            strategy = decorateWithFixedCutStrategy(strategy, statusBarSize, 0);
-        }
-        else
+        switch (strategyName)
```
Comment (uarlouski) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java:63]: should not we allow end users to set strategy name case-insensitively?

Comment (ikalinin1) [reply vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java:63]: I don't think it ever worked this way :)

File: vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java [hunk 2]
```diff
@@ -47,24 +54,31 @@ public MobileAppAshotFactory(ScreenshotCropper screenshotCropper, MobileAppWebDr
...
         {
-            strategy = getStrategyBy(strategyName).getDecoratedShootingStrategy(new MobileViewportShootingStrategy());
-            strategy = downscale ? scaling(strategy, (float) this.getDpr()) : strategy;
+            case SIMPLE:
```
Comment (ikalinin1) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java:65]: Can we make `getStrategyBy(strategyName)` return appropriate strategy?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-visual.adoc [hunk 3]
```diff
@@ -230,6 +230,11 @@ image::example.png[Difference between native/web footer/header image]
 |`true`
 a|Whether to append Android bottom navigation bar (black bar with system buttons on the screenshots below table) or not.
 
+[NOTE]
+====
+The property available only for *VIEWPORT* shooting strategy
```
[the remark this answers is not in the document]
Comment (IBrun) [reply docs/modules/plugins/pages/plugin-visual.adoc:235]: Screenshots taken on failures has independent logic

File: vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java (hunk 2 above)
Comment (IBrun) [reply vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java:65]: Not possible now, as there dependencies on type of mobile OS

File: vividus-plugin-mobile-app/src/main/resources/properties/profile/mobile_app/profile.properties [hunk 4]
```diff
@@ -12,7 +12,7 @@ mobile-environment.real-device=false
 mobile-environment.wda-bundle-id=
 
 mobile.screenshot.strategy=mobile-app
-mobile.screenshot.strategy.mobile-app.shooting-strategy=SIMPLE
+mobile.screenshot.strategy.mobile-app.shooting-strategy=VIEWPORT
 mobile.screenshot.downscale=false
 mobile.screenshot.android.append-bottom-navigation-bar=true
-selenium.screenshot.strategy=SIMPLE
+selenium.screenshot.strategy=VIEWPORT
```
[the remark this answers is not in the document]
Comment (IBrun) [reply vividus-plugin-mobile-app/src/main/resources/properties/profile/mobile_app/profile.properties:18]: It's used here https://github.com/vividus-framework/vividus/blob/9a0eb695514878aa1daafc29ce4a83b43ca7634f/vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java#L53

File: docs/modules/plugins/pages/plugin-visual.adoc [hunk 5]
```diff
@@ -284,8 +289,12 @@ where:
 |The size of right part of the final screenshot to cut.
 
 |`mobile.screenshot.strategy.<YOUR_STRATEGY_NAME>.shooting-strategy`
-|`SIMPLE`
-|Screenshot shooting strategy name (`SIMPLE` by default)
+|`VIEWPORT`, `FULL_SCREEN`, `SIMPLE`
+a|Screenshot shooting strategy name (`VIEWPORT` by default)
+[WARNING]
+====
+The shooting strategy `SIMPLE` is deprecated and will be removed in VIVIDUS 0.6.0. Use `VIEWPORT` instead. `VIEWPORT` is equal to `SIMPLE` shooting strategy
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-visual.adoc:296]: The shooting strategy `SIMPLE` is deprecated and will be removed in VIVIDUS 0.6.0. Use `VIEWPORT` instead. `VIEWPORT` is equal to `SIMPLE` shooting strategy

File: vividus-plugin-mobile-app/src/main/resources/properties/profile/mobile_app/profile.properties (hunk 4 above)
[the remark this answers is not in the document]
Comment (valfirst) [reply vividus-plugin-mobile-app/src/main/resources/properties/profile/mobile_app/profile.properties:18]: > it seems it won't work for web

**Review by ikalinin1 — APPROVED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java [hunk 6]
```diff
@@ -47,24 +54,31 @@ public MobileAppAshotFactory(ScreenshotCropper screenshotCropper, MobileAppWebDr
...
+                break;
+            case "FULL_SCREEN":
+                strategy = ShootingStrategies.simple();
+                break;
+            default:
+                throw new IllegalArgumentException(
+                        String.format("Unknown shooting strategy with the name: %s", strategyName));
         }
 
+        strategy = downscale ? scaling(strategy, (float) this.getDpr()) : strategy;
```
[the remark this answers is not in the document]
Comment (valfirst) [reply vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java:79]: not fixed

File: vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java [hunk 7]
```diff
@@ -47,22 +54,26 @@ public MobileAppAshotFactory(ScreenshotCropper screenshotCropper, MobileAppWebDr
     }
 
     @Override
+    @SuppressFBWarnings("SF_SWITCH_FALLTHROUGH")
     public AShot create(Optional<ScreenshotParameters> screenshotParameters)
     {
-        String strategyName = screenshotParameters.flatMap(ScreenshotParameters::getShootingStrategy)
-                .orElseGet(this::getScreenshotShootingStrategy);
-
+        String strategyName = screenshotParameters.flatMap(ScreenshotParameters::getShootingStrategy).orElse(VIEWPORT);
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/selenium/mobileapp/screenshot/MobileAppAshotFactory.java:60]: that looks like a breaking change

**Review by valfirst — APPROVED** [review]

**Comment by valfirst** [comment]
The changes have been reverted due to failures in system tests. @IBrun please, fix the tests

**Pull request merged**